rotkehlchen/globaldb/handler.py

```diff
@@ -944,6 +946,33 @@ def get_evm_tokens(
 
         return tokens
 
+    @staticmethod
+    def get_token_detection_data(
+            chain_id: ChainID,
+            exceptions: set[ChecksumEvmAddress],
+            protocol: str | None = None,
+    ) -> list[EvmTokenDetectionData]:
+        result = []
+        query = 'SELECT identifier, address, decimals FROM evm_tokens WHERE chain=?'
+        bindings: list[int | str] = [chain_id.serialize_for_db()]
+        if protocol is not None:
+            query += ' AND protocol=?'
+            bindings.append(protocol)
+
+        with GlobalDBHandler().conn.read_ctx() as cursor:
+            cursor.execute(query, bindings)
+            for identifier, address, decimals in cursor:
+                if address in exceptions:
+                    continue
+
+                result.append(EvmTokenDetectionData(
+                    identifier=identifier,
+                    address=address,
+                    decimals=decimals if decimals is not None else DEFAULT_TOKEN_DECIMALS,  # TODO: at least two tokens are missing the decimals in my DB and also the EvmToken class allows decimals to be None. We need to fix it.  # noqa: E501
```

Comment:
what would the fix be here?

Reply:
I was thinking that we would need to enforce the decimals for the EvmTokens but now I'm not 100% sure. We have the NFTs as child of the EvmTokens. In theory all the EvmTokens should have decimals unless I'm missing something.